tests/unit/components/NcAppSettingsDialog/register-unregister.spec.ts

```diff
@@ -0,0 +1,119 @@
+		onMounted(() => register?.(props.id, 'test_name'))
+	},
+})
+
+describe('NcAppSettingsDialog: Sections registration', () => {
+	it('injects register function to children', async () => {
+		const wrapper = mount(NcAppSettingsDialog, {
+			slots: {
+				default: MockSection,
+			},
+			propsData: {
+				open: true,
+			},
+		})
+
+		expect(wrapper.vm.$data.sections).toHaveLength(1)
+		expect(wrapper.vm.$data.sections[0]).toEqual({ id: 'test_id', name: 'test_name' })
+	})
+
+	it('can register a new section', async () => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const wrapper = mount<Vue & { registerSection: any }>(NcAppSettingsDialog, {
+			propsData: {
+				open: true,
+			},
+		})
+
+		wrapper.vm.registerSection('test_id', 'test_name')
+		expect(wrapper.vm.$data.sections).toHaveLength(1)
+		expect(wrapper.vm.$data.sections[0]).toEqual({ id: 'test_id', name: 'test_name' })
+	})
+
+	it('warn on register a already registered section name', async () => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const wrapper = mount<Vue & { registerSection: any }>(NcAppSettingsDialog, {
+			propsData: {
+				open: true,
+			},
+		})
+
+		const spy = jest.spyOn(globalThis.console, 'error')
+		spy.mockImplementationOnce(() => {})
+
+		// First call should be OK
+		wrapper.vm.registerSection('test_id', 'test_name')
+		expect(wrapper.vm.$data.sections).toHaveLength(1)
+		expect(spy).not.toHaveBeenCalled()
+
+		// Second one should unregister first and replace with this one, but show an error
+		wrapper.vm.registerSection('test_id_2', 'test_name')
+		expect(wrapper.vm.$data.sections).toHaveLength(2)
+		expect(spy).toHaveBeenCalled()
+	})
+
+	it('error on register a already registered section ID', async () => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const wrapper = mount<Vue & { registerSection: any }>(NcAppSettingsDialog, {
+			propsData: {
+				open: true,
+			},
+		})
+
+		// First call should be OK
+		wrapper.vm.registerSection('test_id', 'test_name')
+		expect(wrapper.vm.$data.sections).toHaveLength(1)
+		// Second one should unregister first and replace with this one, but show an error
+		expect(() => wrapper.vm.registerSection('test_id', 'test_other_name')).toThrow()
+		expect(wrapper.vm.$data.sections).toHaveLength(1)
+	})
+
+	it('can unregister a section', async () => {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		const wrapper = mount<Vue & { registerSection: any, unregisterSection: any }>(NcAppSettingsDialog, {
+			propsData: {
+				open: true,
+			},
+		})
+
+		wrapper.vm.registerSection('test_id', 'test_name')
+		wrapper.vm.registerSection('test_id2', 'test_name2')
+		expect(wrapper.vm.$data.sections).toHaveLength(2)
+		wrapper.vm.unregisterSection('test_id')
+		expect(wrapper.vm.$data.sections).toHaveLength(1)
+		expect(wrapper.vm.$data.sections[0]).toEqual({ id: 'test_id2', name: 'test_name2' })
+	})
```

Comment:
IMO, we should test the final result here instead.

These tests are much relayed on the component implementation and test their internal implementation.

`vm.$data` should never be tested, it is not a part of a component's contract. Otherwise, we cannot rename `sections` property or refactor component to Composition API without breaking tests.

Although `registerSection` is provided, here it is not a public `provide` but an internal thing used to connect the dialog with the section components supposed to be used as an "organic whole".

We should test that navigation is rendered correctly while the components are used as the end developer uses them.

Reply:
Makes it a little bit more complex as we need to wait for next tick, but I adjusted the tests as I agree with the idea :)